### PR TITLE
Minor updates: see comments below

### PR DESCRIPTION
**Pull request by WilliamBerryiii**
Implemented proper equals for Position struct
Minor code clean up
Began updating naming of properties and fields

### CODE REVIEW

File: Assets/Voxelgon/Ship/ShipManager.cs [hunk 1]
```diff
@@ -1,69 +1,77 @@
...
-        if(Input.GetButtonUp("Kill Rotation")) {
-            killRot = !killRot;
-        }
-
-        if(Input.GetButtonUp("Kill Translation")) {
-            killTrans = !killTrans;
+﻿using System;
+using UnityEngine;
+
+namespace Assets.Voxelgon.Ship
```
Comment (drewcassidy) [inline Assets/Voxelgon/Ship/ShipManager.cs:4]: This should really just be Voxelgon.Ship
Also no newline before opening brace, please

**Comment by drewcassidy** [comment]
Thanks!

The ShipManager and RCSport classes are probably going to get a big overhaul, since I need to make them balance each other on opposite sides of the ship

as for the underscore in private variable names, I'm not sure. I've been trying to adhere the [Official .NET guidelines](https://msdn.microsoft.com/en-us/library/ta31s3bc.aspx) for variable naming

**Comment by WilliamBerryiii** [comment]
Understood on the private fields.  The use of an underscore is no longer a violation of the conventions and is CLS compliant.  I find them particularly useful in the case where you may end up putting a public property over the field or in the instance where the field name may also conflict with a method name ... ie. Foo(); vs private int foo;  See this pretty fair assessment on SO: http://stackoverflow.com/a/17937309/1276028

**Comment by drewcassidy** [comment]
Alright. I have considered the underscores before, and I do sometimes confuse my properties and local variables, so I'll go with it

I do want to use namespaces starting with Voxelgon, not Assets. Unity requires all code to be in the Assets directory, which is why the folders are set up that way in the first place

**Comment by WilliamBerryiii** [comment]
We can set the root name space for each project in the proj files so form there it's up to you if you want Voxelgon.Assets.Foo or just Voxelgon.Foo

**Comment by drewcassidy** [comment]
the  `<RootNamespace>` field isnt even filled in the .csproj file. What whould I add to make it Voxelgon.Foo?

**Comment by drewcassidy** [comment]
Actually, this appears to be something that depends on the IDE you are using. Using OmnisharpSublime, the namespaces are always Voxelgon.Assets.Voxelgon.Foo, which I have to change manually,  and VSCode doesnt fill namespaces automatically at all.

File: Assets/Voxelgon/Math/Position.cs [hunk 2]
```diff
@@ -1,43 +1,67 @@
...
+            unchecked
+            {
+                var hashCode = _x.GetHashCode();
+                hashCode = (hashCode*397) ^ _y.GetHashCode();
+                hashCode = (hashCode*397) ^ _z.GetHashCode();
+                return hashCode;
+            }
+        }
+
+        readonly short _x;
```
Comment (drewcassidy) [inline Assets/Voxelgon/Math/Position.cs:28]: Move variables above methods, see [this](http://stackoverflow.com/questions/150479/order-of-items-in-classes-fields-properties-constructors-methods/310967#310967).

Comment (WilliamBerryiii) [inline Assets/Voxelgon/Math/Position.cs:28]: Will do!

**Comment by WilliamBerryiii** [comment]
some more fixes per code review and additional code clean up.  Merge take care of.  Hopefully line endings work for you too.

**Comment by WilliamBerryiii** [comment]
Don't merge this ... I committed conflicts apparently ... working

**Comment by drewcassidy** [comment]
I'm still not liking the namespace names. Where are you getting those from, Visual Studio? I'll have to find my windows VM and play around with Visual Studio, see if i can get those how I like them

**Comment by WilliamBerryiii** [comment]
I am going to tear this up and try again ... the merge conflicts are next to un-resolvable.

I'll close this and open an issue to discuss the name spaces.

**Comment by drewcassidy** [comment]
sorry, I tried to fix the tabs/spaces thing myself for other files, didnt realize you were going to do the same. I dont know what happened with my config to go back to tabs...

**Comment by WilliamBerryiii** [comment]
no worries ... the changes are easy to work back in!

File: Assets/Voxelgon/Math/Position.cs [hunk 3]
```diff
@@ -1,43 +1,67 @@
...
+            return obj is Position && Equals((Position) obj);
+        }
+
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                var hashCode = _x.GetHashCode();
+                hashCode = (hashCode*397) ^ _y.GetHashCode();
+                hashCode = (hashCode*397) ^ _z.GetHashCode();
```
Comment (drewcassidy) [inline Assets/Voxelgon/Math/Position.cs:23]: Where did the number 397 come from?

Comment (WilliamBerryiii) [inline Assets/Voxelgon/Math/Position.cs:23]: It's just a prime number. See here for why you use them for hashcode
calculations: http://stackoverflow.com/a/263416/1276028 

On 2015-12-17 19:20, Andrew Cassidy wrote: 

> In Assets/Voxelgon/Math/Position.cs [1]: 
> 
> > - return _x == other._x && _y == other._y && _z == other._z;
> > - }
> >   +
> > - public override bool Equals(object obj)
> > - {
> > - if (ReferenceEquals(null, obj)) return false;
> > - return obj is Position && Equals((Position) obj);
> > - }
> >   +
> > - public override int GetHashCode()
> > - {
> > - unchecked
> > - {
> > - var hashCode = _x.GetHashCode();
> > - hashCode = (hashCode*397) ^ _y.GetHashCode();
> > - hashCode = (hashCode*397) ^ _z.GetHashCode();
> 
> Where did the number 397 come from? 
> 
> ## 
> 
> Reply to this email directly or view it on GitHub [2].

## Links:

[1] https://github.com/Voxelgon/Voxelgon/pull/13#discussion_r47990828
[2] https://github.com/Voxelgon/Voxelgon/pull/13/files#r47990828

**Pull request closed without merging**